### PR TITLE
webapi: fix ambiguous options support for dual-specified paths

### PR DESCRIPTION
**Pull request by andrewbonney**
We define OPTIONS support automatically for various API route decorators. Unfortunately, as we often duplicate route definitions with a trailing and non trailing slash variant, in Python 3 Werkzeug ends up with two OPTIONS definitions for some paths, and selects from them at random. This can result in a 308 code being generated which is invalid for OPTIONS.

This patch ensures we only ever define the OPTIONS method once per path, preferring the one without a trailing slash.

I would note that OPTIONS / crossdomain support in WebAPI is still messy and really ought to be handled more consistently, but I think this is a bigger job.

### CODE REVIEW

**Comment by dannymeloy** [comment]
TL;DR - LGTM

This seems like a necessary evil. The way the Web API is constructed, iterating through every route function in the entire inheritance tree, the initial for loop needs to exist to get a full 'picture' of all the routes. Having a uniform set of attribute names would have been more ideal but this is probably as good as it can get
